### DISCUSSION TITLE
Will Carbon fix the Vector name?

### DISCUSSION THREAD

**Discussion by freedxmgxd** [carbon-language/carbon-lang#1628] [General]
I think that Carbon should naming the Dynamic Sized Arrays just 'Array', and the fixed size 'array'. Because of the difference of concepts with the mathematical meaning of the Vectors, and make more sense to me in this way, like the difference at String and &str in Rust.

(I don't read the source code because I'm a junior so some things are very complicated to me, but I hope to help the carbon development soon as possible.)

**Comment by jonmeow** [comment]
I think at one point we were discussing Array and FixedArray, I think it's likely something that'll change but the name hasn't really been decided because we haven't delved too deeply into the common libraries yet.

**Reply by OlaFosheimGrostad** [reply]
Maybe you can call both array… like std::span.
